platforms/documentation/docs/src/docs/userguide/best-practices/bp_general.adoc

```diff
@@ -15,4 +15,36 @@
 [[best_practices_general]]
 = General Gradle Best Practices
 
-Add content here.
+== Do Not Use Internal APIs
+
+Do not use APIs that are inside an `internal` Java package or contain `Internal` or `Impl`
```

Comment:
💅 I think we should be more precise about defining an `internal` package as any package where any of the segments in the package name is `internal`.

Also we should say not just "contains" but has Internal/Impl as a suffix to the name, as there might be edge cases here where this includes valid public types.

Reply:
Used [lkasso](https://github.com/lkasso)'s suggestion for this. Let me know if the updated version is better.